src/FieldWarning/Assets/Units/ShellBehaviour.cs

```diff
@@ -118,33 +126,61 @@ private void Update()
             Vector3 worldForward = transform.TransformDirection(Vector3.forward);
             worldForward = new Vector3(worldForward.x, 0, worldForward.z);
             Vector3 translation = _forwardSpeed * worldForward * Time.deltaTime
-                                  + _verticalSpeed * Vector3.up * Time.deltaTime;
-            transform.LookAt(transform.position + translation);
+                                  + _verticalSpeed * Vector3.up * Time.deltaTime
+                                  + _targetVelocity * Time.deltaTime; /// * Constants.MAP_SCALE;
+            ///transform.LookAt(transform.position + translation);
             transform.Translate(
                     translation,
                     Space.World);
 
             _verticalSpeed -= GRAVITY * Time.deltaTime;
 
-
-            // small trick to detect if shell has reached the target
-            float distanceToTarget = Vector3.Distance(transform.position, _targetCoordinates);
-            if (distanceToTarget > _prevDistanceToTarget)
+            if (_justLaunched)
             {
-                transform.position = _targetCoordinates;
-                Explode();
+                print("instant collision not avoided");
             }
-            _prevDistanceToTarget = distanceToTarget;
-
-            //transform.rotation.SetLookRotation(translation);
+            _justLaunched = false;
         }
 
         private void OnTriggerEnter(Collider other)
         {
-            if (other.GetComponent<CaptureZone>() == null)
+            if (_justLaunched)
             {
-                Explode();
+                _launchPlatform = other;
+                _justLaunched = false;
+                print("instant collision avoided now");
+            }
+            else if(_launchPlatform != other)
+            {
+                UnitDispatcher target = other.gameObject.GetComponentInParent<UnitDispatcher>();
+
+                if (target != null && !_ammo.IsAoe)
+                {
+                    Kinetic(target);
```

Comment:
It would be much better if we apply the damage on the owner of the collider, or do an area lookup or something. Won't the current code damage the intended target even if the shell hits a completely different unit or building?

Reply:
That is what it does, applies damage to the owner of the collider. This is in the scope of the collision event and target is the owner of the collider from that event.